packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -214,113 +218,52 @@ const useTransactionHistoryWithoutStatuses = (
     cctpTransfersTestnet.isLoadingDeposits ||
     cctpTransfersTestnet.isLoadingWithdrawals
 
-  const shouldFetchNextPageForChainPair = useCallback(
-    (chainPairIndex: number, direction: 'deposits' | 'withdrawals') => {
-      const isDeposits = direction === 'deposits'
-      const page = isDeposits ? depositsPage : withdrawalsPage
-      const transactions = isDeposits ? deposits : withdrawals
-
-      if (page === 0) {
-        return true
-      }
-
-      const txCountForChainPair = transactions[chainPairIndex]?.length ?? 0
-      // fetch next page for a chain pair if all fetched pages are full
-      // we check for >= in case some txs were included by initiating a transfer
-      return txCountForChainPair / page >= PAGE_SIZE
-    },
-    [depositsPage, withdrawalsPage, deposits, withdrawals]
-  )
-
-  const { data: depositsData, error: depositsError } = useSWRImmutable(
-    address ? ['tx_list', 'deposits', address, depositsPage] : null,
-    ([, , _address, _page]) => {
+  const {
+    data: depositsData,
+    error: depositsError,
+    isLoading: depositsLoading
+  } = useSWRImmutable(
+    address ? ['tx_list', 'deposits', address] : null,
+    ([, , _address]) => {
       return Promise.all(
-        multiChainFetchList.map((chainPair, chainPairIndex) => {
-          if (!shouldFetchNextPageForChainPair(chainPairIndex, 'deposits')) {
-            return []
-          }
-
+        multiChainFetchList.map(chainPair => {
           return fetchDeposits({
             sender: _address,
             receiver: _address,
             l1Provider: getProvider(chainPair.parentChain),
             l2Provider: getProvider(chainPair.chain),
-            pageNumber: _page,
-            pageSize: PAGE_SIZE
+            pageNumber: 0,
+            pageSize: 1000
           })
         })
       )
     }
   )
 
-  const { data: withdrawalsData, error: withdrawalsError } = useSWRImmutable(
-    address ? ['tx_list', 'withdrawals', address, withdrawalsPage] : null,
-    ([, , _address, _page]) => {
+  const {
+    data: withdrawalsData,
+    error: withdrawalsError,
+    isLoading: withdrawalsLoading
+  } = useSWRImmutable(
+    address ? ['tx_list', 'withdrawals', address] : null,
+    ([, , _address]) => {
       return Promise.all(
-        multiChainFetchList.map((chainPair, chainPairIndex) => {
-          if (!shouldFetchNextPageForChainPair(chainPairIndex, 'withdrawals')) {
-            return []
-          }
-
+        multiChainFetchList.map(chainPair => {
           return fetchWithdrawals({
             sender: _address,
             receiver: _address,
             l1Provider: getProvider(chainPair.parentChain),
             l2Provider: getProvider(chainPair.chain),
-            pageNumber: _page,
-            pageSize: PAGE_SIZE
+            pageNumber: 0,
+            pageSize: 1000
```

Comment:
It's much simpler to fetch 1000 rather than looping. We can implement it if we need to in the future but `useSWRImmutable` is more reliable than `Infinite` if we can use that.

1000 txs should **per chain pair** should be more than enough for pretty much everyone.

Reply:
It also simplifies a lot of logic